src/types/tuple.rs

```diff
@@ -691,14 +691,14 @@ fn type_output() -> TypeInfo {
         fn extract_bound(obj: &Bound<'py, PyAny>) -> PyResult<Self>
         {
             let t = obj.downcast::<PyTuple>()?;
-            if t.len() == $length {
+            if t.len() != $length {
+                Err(wrong_tuple_length(t, $length))
+            } else {
                 #[cfg(any(Py_LIMITED_API, PyPy, GraalPy))]
                 return Ok(($(t.get_borrowed_item($n)?.extract::<$T>()?,)+));
 
                 #[cfg(not(any(Py_LIMITED_API, PyPy, GraalPy)))]
                 unsafe {return Ok(($(t.get_borrowed_item_unchecked($n).extract::<$T>()?,)+));}
-            } else {
-                Err(wrong_tuple_length(t, $length))
```

Comment:
This lint looks like a false positive to me.

Something we could consider changing here would be to remove the `return` statements and use implicit return. But with the two `cfg` options, I'm not certain that'd be more readable.

Reply:
I had tried that initially, but explicit return seems to be required:

```
697 |                 #[cfg(any(Py_LIMITED_API, PyPy, GraalPy))]
    |                 ------------------------------------------ only `;` terminated statements or tail expressions are allowed after this attribute
698 |                 Ok(($(t.get_borrowed_item($n)?.extract::<$T>()?,)+))
    |                                                                     ^ expected `;` here
```